gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -393,28 +393,41 @@ def prop_defined(name, cls, rules={})
     handle_redaction_option(name, rules[:redaction]) if rules[:redaction]
   end
 
-  # checked(:never) - Rules hash is expensive to check
+    # checked(:never) - Rules hash is expensive to check
   sig {params(name: Symbol, rules: Rules).void.checked(:never)}
   private def define_getter_and_setter(name, rules)
     T::Configuration.without_ruby_warnings do
       if !rules[:immutable]
         if method(:prop_set).owner != T::Props::Decorator
-          @class.send(:define_method, "#{name}=") do |val|
-            T.unsafe(self.class).decorator.prop_set(self, name, val, rules)
-          end
+          @class.class_eval <<~SETTER
+            def #{name}=(val)
+              rules = T.unsafe(self.class).decorator.prop_rules(:#{name})
+              T.unsafe(self.class).decorator.prop_set(self, :#{name}, val, rules)
```

Comment:
pretty sure these `T.unsafe` were to get the `decorator.rb` file itself to typehceck. now that you're not needing to type check it, you can remove the `T.unsafe`.